File: week7/tutorial/problem3_template.scala
package comp9313.lab7
import org.apache.spark.sql.SparkSession
import org.apache.spark.sql.functions._
import org.apache.spark.sql.DataFrame

object VotesStat {

  def Question2(textFile: DataFrame, spark: SparkSession) {
    import spark.implicits._
    // write your code here
  }
  
  def Question1(textFile: DataFrame, spark: SparkSession) {
    import spark.implicits._
    //write your code here  
  }


  def main(args: Array[String]) {
    val inputFile = args(0)
    val outputFolder = args(1)
    val spark = SparkSession.builder.appName("VotesStat").getOrCreate()
    val textFile = spark.read.csv(inputFile).toDF("Id", "PostId", "VoteTypeId", "UserId", "CreationTime")
    
    Question1(textFile, spark)
    Question2(textFile, spark)
    spark.stop()
  }
}

File: week7/tutorial/problem3_template.py
from pyspark.sql.session import SparkSession
import sys

class Problem3:

    def Question1(self, fields):
        # add your code here

    
    def Question2(self, fields):
    	# add your code here

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Wrong inputs")
        sys.exit(-1)

    inputFile = sys.argv[1]
    spark = SparkSession.builder.master("local").appName("problem3").getOrCreate()
    fields = spark.read.csv(inputFile).toDF("Id", "PostId", "VoteTypeId", "UserId", "CreationTime")

    p3 = Problem3()
    p3.Question1(fields)
    p3.Question2(fields)
    spark.stop()






